libc/cmake/modules/LLVMLibCTestRules.cmake

```diff
@@ -561,7 +561,7 @@ function(add_integration_test test_name)
   # to expand the list (by including the option COMMAND_EXPAND_LISTS). This
   # makes `add_custom_target` construct the correct command and execute it.
   set(test_cmd
-      ${INTEGRATION_TEST_ENV}
+      ${CMAKE_CROSSCOMPILING_EMULATOR} ${INTEGRATION_TEST_ENV}
```

Comment:
We do the following for the GPU build.
```shell
$> clang main.c --target=amdgcn-amd-amdhsa -mcpu=gfx1030 crt1.o -lc -o exe
$> ./emulator <optional arguments> exe
```
This looks very similar to cross-compiling by design. I'm thinking it would be good to unify this approach.

Reply:
That sounds like a good idea, but how do you handle env variables in these cases? 

Currently, the script I wrote expects `<env variables> <full path to binary> <args>` to work properly... We need the full path to the binary to scp any test data present in the binary directory; we then copy the binary to a temp dir in the qemu-system and call `ssh foo@bar <env variables> ~/<temp dir>/<binary> <args>`.

(I should probably upload the script I'm using somewhere).

Reply:
I don't use any env vars, but presumably it would be handled just by the loader. E.g. if I do `env foo ./amdhsa_loader <args> <image>` it would read `foo` from the environment and pass it to the `image` it launches.

Reply:
do you know does it detects the foo env variable in the loader? Or will it just send every env var on the system to the image?

Reply:
Every single one, it clones the current user environment.

Reply:
https://github.com/llvm/llvm-project/blob/main/libc/utils/gpu/loader/Loader.h#L92 here's the code for it, pretty much just uses the `envp` passed in from the `main` function.

Reply:
that actually makes things easier for me.

What do you think about dropping `gpu_loader_exe` and using just `CMAKE_CROSSCOMPILING_EMULATOR` instead?

Reply:
Would this be an argument passed in the `cmake` invocation? Because the `gpu_loader_exe` is built internally so it's not known until CMake runs. We could potentially just set `CMAKE_CROSSCOMPILING_EMULATOR` manually?

Reply:
I just ran some tests here and it works both ways: either you set it during the cmake config via cmd line or you can set it using `set(CMAKE_CROSSCOMPILING_EMULATOR "foobar")`.

My only concern is that it doesn't tell the user that it overwrote an argument passed via cmd line, but I guess we can always throw a warning there.

I'll update this PR to put `$CMAKE_CROSSCOMPILING_EMULATOR` right after the `$gpu_loader_exe` for now, and send all environment variables as the loader does. If you can test using only `$CMAKE_CROSSCOMPILING_EMULATOR` and it does work, we can send another PR to remove `$gpu_loader_exe` later.

Reply:
SG, I'll test it once this lands.